sql/src/test/java/org/apache/apex/malhar/sql/FileEndpointTest.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.sql;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.TimeZone;
+
+import javax.validation.ConstraintViolationException;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.apex.malhar.sql.table.CSVMessageFormat;
+import org.apache.apex.malhar.sql.table.FileEndpoint;
+
+import org.apache.commons.io.FileUtils;
+
+import org.apache.hadoop.conf.Configuration;
+
+import com.google.common.collect.ImmutableList;
+
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.LocalMode;
+import com.datatorrent.api.StreamingApplication;
+
+public class FileEndpointTest
+{
+  private TimeZone defaultTZ;
+
+  public static String apex_concat_str(String s1, String s2)
+  {
+    return s1 + s2;
+  }
+
+  @Before
+  public void setUp() throws Exception
+  {
+    defaultTZ = TimeZone.getDefault();
+    TimeZone.setDefault(TimeZone.getTimeZone("GMT"));
+
+    File inputFile = new File("/tmp/input/input.csv");
+    FileUtils.writeLines(inputFile, ImmutableList.of("15/02/2016 10:15:00 +0000,1,paint1,11",
+        "15/02/2016 10:16:00 +0000,2,paint2,12",
+        "15/02/2016 10:17:00 +0000,3,paint3,13", "15/02/2016 10:18:00 +0000,4,paint4,14",
+        "15/02/2016 10:19:00 +0000,5,paint5,15", "15/02/2016 10:10:00 +0000,6,abcde6,16"));
+  }
+
+  @After
+  public void tearDown() throws Exception
+  {
+    FileUtils.deleteDirectory(new File("/tmp/input"));
+    FileUtils.deleteDirectory(new File("/tmp/output"));
+
+    TimeZone.setDefault(defaultTZ);
+  }
+
+  @Test
+  public void testApplication() throws Exception
+  {
+    File modelFile = new File("src/test/resources/model/model_file_csv.json");
+    String model = FileUtils.readFileToString(modelFile);
+
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      Configuration conf = new Configuration(false);
+      lma.prepareDAG(new Application(model), conf);
+      LocalMode.Controller lc = lma.getController();
+      lc.run(10 * 1000); // runs for 30 seconds and quits
```

Comment:
This emits to console output operator. Hence nothing to verify here. Will do it for another test.